### ISSUE TITLE
Sometimes, RAPI does not return a tuple in opresult

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#674]
Originally reported of Google Code with ID 618.

```
<b>What steps will reproduce the problem?</b>
Some errors delivered through opresult are formatted incorrectly; they do not include the error type plus the error text. Instead, they include just the error text. 

Change 599ee321eb778eaef5020a1eae1e4c8adf7b50ae from 2010 sounds like it should have fixed this.

Used version: 2.6.2

<b>What is the expected output? What do you see instead?</b>
Based on 599ee321eb, expected output:

   {'end_ts': [1382587208, 129359],
    'id': '468649',
    'oplog': [[]],
    'opresult': ['OpExecError', ['Unclean master daemon shutdown']],
    'ops': [{'OP_ID': 'OP_INSTANCE_REBOOT',
             'comment': None,
             'debug_level': 0,
             'depends': None,
             'dry_run': False,
             'ignore_secondaries': False,
             'instance_name': 'XYZ',
             'priority': 0,
             'reboot_type': 'full',
             'shutdown_timeout': 120}],
    'opstatus': ['error'],
    'received_ts': [1382585847, 942769],
    'start_ts': [1382585848, 68114],
    'status': 'error',
    'summary': ['INSTANCE_REBOOT(XYZ)']}

Received output:

   {'end_ts': [1382587208, 129359],
    'id': '468649',
    'oplog': [[]],
    'opresult': ['Unclean master daemon shutdown'],
    'ops': [{'OP_ID': 'OP_INSTANCE_REBOOT',
             'comment': None,
             'debug_level': 0,
             'depends': None,
             'dry_run': False,
             'ignore_secondaries': False,
             'instance_name': 'XYZ',
             'priority': 0,
             'reboot_type': 'full',
             'shutdown_timeout': 120}],
    'opstatus': ['error'],
    'received_ts': [1382585847, 942769],
    'start_ts': [1382585848, 68114],
    'status': 'error',
    'summary': ['INSTANCE_REBOOT(XYZ)']}


We've run into this problem only occasionally; the output similar to the above is the primary information we have about this problem.
```
Originally added on 2013-11-21 13:57:11 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
It seems that lines 1675/1676 in jqueue.py are to blame:

job.MarkUnfinishedOps(constants.OP_STATUS_ERROR,
                      "Unclean master daemon shutdown")

Instead, this should be something like:

daemon_error = errors.OpExecError("Unclean master daemon shutdown")
job.MarkUnfinishedOps(constants.OP_STATUS_ERROR,
                      _EncodeOpError(daemon_error))

Once I can reproduce and can confirm this fixes things, I'll make a patch for this.
I don't know if this is a big enough bug to warrant a 2.6 release, but there are some other 2.6 issues pending so we'll see. Thanks for reporting this!
```
Originally added on 2013-11-21 14:38:37 +0000 UTC.

Added Labels: Type-Defect Priority-Medium
Changed State: Accepted
Added to Milestone: Release2.6

**Comment by Ganeti-Issues-Migrator** [comment]
```
Fixed, but pushed to stable-2.8 as a new 2.6 version is not likely to be released soon.

Fixed in commit:

commit 9cbcb1beeb080d69fe9591cda96399cbc66eaa56
Author: Hrvoje Ribicic <riba@google.com>
Date:   Thu Dec 5 10:49:01 2013 +0100

    Fix job error message after unclean master shutdown
    
    According to commit 599ee321eb, any job-related error messages should
    be encoded within a Ganeti-specific error and not passed on as a
    string, to allow for easier parsing.
    
    For jobs suffering from an undesirable status after an unclean master
    daemon shutdown, the message was not encoded, as reported in issue 618.
    This patch fixes the problem.
    
    Signed-off-by: Hrvoje Ribicic <riba@google.com>
    Reviewed-by: Helga Velroyen <helgav@google.com>

```
Originally added on 2013-12-05 13:17:32 +0000 UTC.

Mentioned Issue with GoogleCode ID 618 has been migrated to GitHub with ID #674.
Changed State: Fixed
Added to Milestone: Release2.8
Removed from Milestone: Release2.6

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-12-13 16:18:23 +0000 UTC.

Changed State: Released
